src/k8s/pkg/utils/control/retry_test.go

```diff
@@ -0,0 +1,64 @@
+package control
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+)
+
+func TestRetryFor(t *testing.T) {
+	t.Run("Retry succeeds", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		retryCount := 3
+		count := 0
+
+		err := RetryFor(ctx, retryCount, 50*time.Millisecond, func() error {
+			count++
+			if count < retryCount {
+				return errors.New("failed")
+			}
+			return nil
+		})
+
+		if err != nil {
+			t.Errorf("Expected nil error, got: %v", err)
+		}
+		if count != retryCount {
+			t.Errorf("Expected retry count %d, got: %d", retryCount, count)
+		}
+	})
+
+	t.Run("Retry fails with context cancellation", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+		defer cancel()
+
+		retryCount := 3
+
+		err := RetryFor(ctx, retryCount, time.Second, func() error {
+			time.Sleep(200 * time.Millisecond)
```

Comment:
i see why this is here, but in general we should avoid having `time.Sleep()` or unit tests that are based on time.

Reply:
ack